pkg/aws/elasticache.go

```diff
@@ -0,0 +1,137 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/elasticache"
+	"github.com/aws/aws-sdk-go/service/elasticache/elasticacheiface"
+	"github.com/aws/aws-sdk-go/service/resourcegroupstaggingapi"
+	"github.com/aws/aws-sdk-go/service/resourcegroupstaggingapi/resourcegroupstaggingapiiface"
+	"github.com/integr8ly/cluster-service/pkg/clusterservice"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"strings"
+)
+
+var _ ActionEngine = &ElasticacheEngine{}
+var TagFilters []*resourcegroupstaggingapi.TagFilter
+
+type ElasticacheEngine struct {
+	elasticacheClient elasticacheiface.ElastiCacheAPI
+	taggingClient     resourcegroupstaggingapiiface.ResourceGroupsTaggingAPIAPI
+	logger            *logrus.Entry
+}
+
+func NewDefaultElastiCacheEngine(session *session.Session, logger *logrus.Entry) *ElasticacheEngine {
+	return &ElasticacheEngine{
+		elasticacheClient: elasticache.New(session),
+		taggingClient:     resourcegroupstaggingapi.New(session),
+		logger:            logger.WithField("engine", "aws_elasticache"),
+	}
+}
+
+func (r *ElasticacheEngine) GetName() string {
+	return "AWS elasticache Engine"
+}
+
+//Delete all RDS resources for a specified cluster
+func (r *ElasticacheEngine) DeleteResourcesForCluster(clusterId string, tags map[string]string, dryRun bool) ([]*clusterservice.ReportItem, error) {
+	logger := r.logger.WithFields(logrus.Fields{"clusterId": clusterId, "dryRun": dryRun})
+	logger.Debug("deleting resources for cluster")
+
+	var reportItems []*clusterservice.ReportItem
+	var replicationGroupsToDelete []string
+	resourceInput := &resourcegroupstaggingapi.GetResourcesInput{
+		ResourceTypeFilters: aws.StringSlice([]string{"elasticache:cluster"}),
+		TagFilters: []*resourcegroupstaggingapi.TagFilter{
+			{
+				Key: aws.String(tagKeyClusterId),
+				Values: aws.StringSlice([]string{
+					clusterId,
+				}),
+			},
+		},
+	}
+	resourceOutput, err := r.taggingClient.GetResources(resourceInput)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to Get resources for clusterID : %s", clusterId)
```

Comment:
Let's replace instances of `errors#Wrapf` with the internal `errors#WrapLog` [1], example here [2]

The end result should be `make vendor/fix` removing the external `errors` package. As the internal implementation uses `fmt#Errorf` under the hood.

[1] https://github.com/integr8ly/cluster-service/blob/master/pkg/errors/errors.go#L10
[2] https://github.com/integr8ly/cluster-service/blob/95cd3421a83db0e95dfdfd00db9afe5c2d42fbb3/pkg/aws/rds.go#L37